inductiva/fluids/scenarios/coastal_area/bathymetry.py

```diff
@@ -329,3 +335,43 @@ def plot(
 
         else:
             return ax
+
+    def to_uniform_grid(self, x_size: int = 200, y_size: int = 200):
+        """Converts the bathymetry to a uniform grid.
+
+        The bathymetry is interpolated to a uniform grid. The grid is defined
+        by the ranges of x and y values, and the number of points in each
+        direction is configurable via the `x_size` and `y_size` arguments.
+
+        Args:
+            x_size: Number of grid points in the x direction.
+            y_size: Number of grid points in the y direction.
+        """
+
+        logging.info(
+            "Interpolating the bathymetry to a uniform grid...\n"
+            "- grid size %d x %d", x_size, y_size)
+
+        # Create uniform grid for interpolation.
+        (x_grid, y_grid) = inductiva.utils.grids.get_meshgrid(
+            x_range=self.x_range,
+            y_range=self.y_range,
+            x_num=x_size,
+            y_num=y_size,
+        )
+
+        depths_grid = inductiva.utils.interpolation.interpolate_to_uniform_grid(
+            x=(self.x, self.y),
+            values=self.depths,
+            x_grid=(x_grid, y_grid),
+        )
+
+        if np.sum(np.isnan(depths_grid)) > 0:
```

Comment:
Could it occur that there are holes in the middle of the bathymetry?

Reply:
Yes, it could!

Reply:
Then, the message below needs to reflect that :)

Reply:
Pardon. It couldn't! There can be holes in the arbitrary $x$ and $y$ coordinate, but we can always interpolate in regions like such in the middle of the bathymetry. The issue is with interpolating at the edges, where we don't have values on both sides of the target locations to interpolate from.

Reply:
Ah ok! Then, it is fine 👍